File: app/src/main/java/com/baronzhang/android/weather/view/home/HomePageContract.java
package com.baronzhang.android.weather.view.home;

import com.baronzhang.android.weather.model.db.entities.minimalist.Weather;
import com.baronzhang.android.weather.view.main.BasePresenter;
import com.baronzhang.android.weather.view.main.BaseView;

/**
 * @author baronzhang (baron[dot]zhanglei[at]gmail[dot]com)
 */
public interface HomePageContract {

    interface View extends BaseView<Presenter> {

        void displayWeatherInformation(Weather weather);
    }

    interface Presenter extends BasePresenter {

        void loadWeather(String cityId, boolean refreshNow);
    }
}

File: weather-mvp-dagger-kotlin/src/main/java/com/minimalist/weather/kotlin/view/contract/HomePageContract.kt
package com.minimalist.weather.kotlin.view.contract


import com.minimalist.weather.kotlin.model.data.entity.weather.Weather

interface HomePageContract {

    interface View : BaseView {

        fun displayWeatherInformation(weather: Weather)
    }

    interface Presenter : BasePresenter<View> {

        fun loadWeather(cityId: String, refreshNow: Boolean)
    }
}
